src/Validator.ts

```diff
@@ -15,7 +15,11 @@ export class Validator {
         return this.validateAsClass(target, target.constructor as AnyClass, options);
     }
 
-    validateAsClass<T>(target: T, clz: AnyClass, options?: ValidationOptions): ValidationResult<T> {
+    validateAsClass<TInput, TExpected>(
+        target: TInput,
+        clz: ClassOf<TExpected>,
+        options?: ValidationOptions,
+    ): ValidationResult<TInput & Partial<TExpected>> {
```

Comment:
I'll try to explain the motivation for changing the return type for a validation.

`TInput` will not necessarily match the class used for validation, `TExpected`.  This is because `TExpected` can have _optional_ properties that `TInput` may not include, but those optional properties may have default values, thus the result may be an extension of `TInput`.

But thinking about this more now, I realise that the result may not include everything in `TInput`, as `TInput` may include properties that exist in `TExpected`, but have been decorated with `Strip`, meaning the validator will exclude them from the result. So really the result should probably be `Partial<TInput> & Partial<TExpected>`. Would you agree?

In hindsight, this should probably be considered a breaking change I think.

Reply:
If you agree this `Partial<TInput> & Partial<TExtended>` is more correct, and that it should be a breaking change, what we could do is create 1.0.0 branch and merge breaking change stuff into it, until we're ready to release that. Let me know your thoughts